adagrams/game.py

```diff
@@ -1,11 +1,134 @@
-def draw_letters():
-    pass
+import random
+
+def draw_letters(): # Time complexity: O(1)  Space complexity: O(1)
+    list_letters_pool = ["A", "A", "A", "A", "A", "A", "A", "A", "A",
+                    "B", "B",
+                    "C", "C",
+                    "D", "D", "D", "D",
+                    "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E",
+                    "F", "F", 
+                    "G", "G", "G", 
+                    "H", "H", 
+                    "I", "I", "I", "I", "I", "I", "I", "I", "I",
+                    "J", 
+                    "K", 
+                    "L", "L", "L", "L",
+                    "M", "M",
+                    "N", "N", "N", "N", "N", "N",
+                    "O", "O", "O", "O", "O", "O", "O", "O",
+                    "P", "P",
+                    "Q", 
+                    "R", "R", "R", "R", "R", "R", 
+                    "S", "S", "S", "S",
+                    "T", "T", "T", "T", "T", "T",
+                    "U", "U", "U", "U",
+                    "V", "V",
+                    "W", "W",
+                    "X", 
+                    "Y", "Y",
+                    "Z"]
+    list_letters_drawn = []
+    for i in range (10):
+        random_letter = random.choice(list_letters_pool)
+        list_letters_drawn.append(random_letter)
+        # remove the drawn letter from the original list of letter pool after 
+        # each random draw
+        list_letters_pool.remove(random_letter)  
+    return list_letters_drawn
+
+def uses_available_letters(word, letter_bank):  
+    # Time complexity: O(n)  Space complexity: O(n)
+    # This portion of function is to count the frequency of each letter drawn 
+    # and store it in dict
+    letter_bank_dict = {}
+    for each_letter in letter_bank:
+        if each_letter not in letter_bank_dict:
+            letter_bank_dict[each_letter] = 1
+        else:
+            letter_bank_dict[each_letter] += 1
+    
+    # This portion of function is to split each char in the word and count the 
+    # frequency of each char. 
+    # If the char is in the dictionary and its count is not over the its 
+    # corresponding count of being drawn.
+    # The loop will skip and continue to check the next char. Once there is a 
+    # char does not fit any of these two conditions, it will return False. 
+    for each_char in word:
+        char_count = word.count(each_char)
+        if each_char in letter_bank_dict and char_count <= \
+        letter_bank_dict[each_char]:
+            continue
+        else:    
+            return False
```

Comment:
while this is valid, `continue` statements aren't commonly used. We can dry this up by doing something like:

```suggestion
        if each_char not in letter_bank_dict or char_count > \
        letter_bank_dict[each_char]:
            return False
```